File: APP03022-mrch.backend.somx.fiscal-api/src/main/java/com/sodimac/fiscal/api/exception/GenericException.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.sodimac.fiscal.api.exception;

/**
 *
 * @author ggalvan
 */
public class GenericException extends Exception {

    private static final long serialVersionUID = -7051709488615576604L;
    private final int code;

    public GenericException(String description, int code) {
        super(description);
        this.code = code;
    }

    public int getCode() {
        return code;
    }

}

File: APP03022-mrch.backend.somx.util-api/src/exceptions/GenericException.ts
export class GenericException extends Error {
    public readonly code: number;

    constructor(code: number, description: string) {
        super(description);
        this.name = 'GenericException';
        this.code = code;

        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, GenericException);
        }
    }
}

export default GenericException;

